### PR TITLE
feat: Create a API class for the AI

### PR DESCRIPTION
**Pull request by Marius-P1**
I've added the basic API to connect the AI to our server and the first exchange (see below).

For now, I've linked the API to the main to let you see how the API works, but it will be put into an AI class in the future.

:exclamation: Feel free to give your opinion on how I documented each function.

image

### CODE REVIEW

File: ai/src/main.py
Comment (Thomaltarix) [inline ai/src/main.py:1]: For me, you should print the error messages in the error output :+1:

File: ai/src/network/API.py
Comment (Thomaltarix) [inline ai/src/network/API.py:1]: You can build your own Exception classes to be more accurate in your errors :+1:  (like in c++)

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
A little request and a advice, but overall it's a great job

**Review by mathieurobert1 — APPROVED** [review]
Seems good :+1:

**Review by Njord201 — DISMISSED** [review]

**Review by AxelF44 — DISMISSED** [review]

**Comment by Marius-P1** [comment]
I added some tests _(before you ask API is complicated to test because it's about the connection between the AI and the server)_

:warning: For the moment you can't have the coverage rapport (like behind) without using a virtual env and the required dependencies, i will make a script later and a readme that explain everything (but i don't think that the priority for the moment and i can evolve in the futur so it's rather useless at this moment to do one)

Coverage :
image

**Review by Thomaltarix — APPROVED** [review]
Clean ! But don't forget to make a README.md for the tests when it will be implemented :+1:

**Comment by theophile-jr** [comment]
Don't forget to follow branches name convention ;) SEEMS GOOD !

**Review by mathieurobert1 — APPROVED** [review]
Seems good :+1:

**Pull request merged**